nifi/nifi-nar-bundles/nifi-flume-bundle/nifi-flume-processors/src/test/java/org/apache/nifi/processors/flume/FlumeSinkProcessorTest.java

```diff
@@ -0,0 +1,165 @@
+        }
+
+        // class doesn't implement Sink
+        results = new HashSet<>();
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, AvroSource.class.getName());
+        runner.enqueue(new byte[0]);
+        pc = runner.getProcessContext();
+        if (pc instanceof MockProcessContext) {
+            results = ((MockProcessContext) pc).validate();
+        }
+        Assert.assertEquals(1, results.size());
+        for (ValidationResult vr : results) {
+            logger.error(vr.toString());
+            Assert.assertTrue(vr.toString().contains("is invalid because unable to create sink"));
+        }
+
+        results = new HashSet<>();
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, NullSink.class.getName());
+        runner.enqueue(new byte[0]);
+        pc = runner.getProcessContext();
+        if (pc instanceof MockProcessContext) {
+            results = ((MockProcessContext) pc).validate();
+        }
+        Assert.assertEquals(0, results.size());
+    }
+
+
+    @Test
+    public void testNullSink() throws IOException {
+        TestRunner runner = TestRunners.newTestRunner(FlumeSinkProcessor.class);
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, NullSink.class.getName());
+        FileInputStream fis = new FileInputStream("src/test/resources/testdata/records.txt");
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put(CoreAttributes.FILENAME.key(), "records.txt");
+        runner.enqueue(fis, attributes);
+        runner.run();
+        fis.close();
+    }
+
+    @Test
+    public void testBatchSize() throws IOException {
+        TestRunner runner = TestRunners.newTestRunner(FlumeSinkProcessor.class);
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, NullSink.class.getName());
+        runner.setProperty(FlumeSinkProcessor.BATCH_SIZE, "1000");
+        runner.setProperty(FlumeSinkProcessor.FLUME_CONFIG,
+            "tier1.sinks.sink-1.batchSize = 1000\n");
+        for (int i = 0; i < 100000; i++) {
+          runner.enqueue(String.valueOf(i).getBytes());
+        }
+        runner.run();
+    }
+    
+    @Test
+    public void testHdfsSink() throws IOException {
+        File destDir = new File("target/hdfs");
+        if (destDir.exists()) {
+          FileUtils.deleteFilesInDir(destDir, null, logger);
+        } else {
+          destDir.mkdirs();
+        }
+
+        TestRunner runner = TestRunners.newTestRunner(FlumeSinkProcessor.class);
+        runner.setProperty(FlumeSinkProcessor.SINK_TYPE, "hdfs");
+        runner.setProperty(FlumeSinkProcessor.FLUME_CONFIG,
+            "tier1.sinks.sink-1.hdfs.path = " + destDir.toURI().toString() + "\n" +
+            "tier1.sinks.sink-1.hdfs.fileType = DataStream\n" +
+            "tier1.sinks.sink-1.hdfs.serializer = TEXT\n" +
+            "tier1.sinks.sink-1.serializer.appendNewline = false"
+        );
+        FileInputStream fis = new FileInputStream("src/test/resources/testdata/records.txt");
```

Comment:
Please use getResourceAsStream so that this test resource will be loaded out of maven target/. This will allow us to later add pre-processing if necessary.

i.e.

InputStream is = getClass().getResourceAsStream("/testdata/records.txt");